packages/backend-defaults/src/entrypoints/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -75,15 +77,18 @@ export const rootHttpRouterServiceFactory = createServiceFactory(
       config: coreServices.rootConfig,
       rootLogger: coreServices.rootLogger,
       lifecycle: coreServices.rootLifecycle,
+      health: coreServices.rootHealth,
     },
-    async factory({ config, rootLogger, lifecycle }) {
+    async factory({ config, rootLogger, lifecycle, health }) {
       const { indexPath, configure = defaultConfigure } = options ?? {};
       const logger = rootLogger.child({ service: 'rootHttpRouter' });
       const app = express();
 
       const router = DefaultRootHttpRouter.create({ indexPath });
       const middleware = MiddlewareFactory.create({ config, logger });
       const routes = router.handler();
+
+      const healthRouter = createHealthRouter({ health });
```

Comment:
I don't think that this file should have any changes at all.

The `rootHealthServiceFactory` should have a dependency on `coreServices.rootHttpRouter` and just call `.use` on it with the right path.

Your `DefaultRootHealthService` could have a `create` method that accepts the root http router service as an argument and does that work. You could also make your `rootHealthServiceFactory` be of the options callback form, where you can pass in custom liveness and readiness callbacks as an adopter, and then the `DefaultRootHealthService` uses those instead of the default ones if they were given.

Reply:
I believe this approach simplifies matters in the event that the adopter wishes to provide their own implementation of the health service. With your approach, they would essentially have to replicate what the default implementation of the healthServiceFactory is doing, remembering the path where the middleware should be added. This could potentially lead to problems if they attach a middleware to a non-standard path.

This alternative approach simplifies things: they only need to provide their own implementation of the `getLiveness` and `getReadiness` methods. No raw operations on the root router are necessary.

Reply:
What I meant was,

```ts
backend.add(
  rootHealthServiceFactory({
    getLiveness: async () => ...,
  })
)
```

as the first level of adaptation, and if you need more,

```ts
backend.add(
  createServiceFactory({
    service: coreServices.rootHealth,
    deps: {
      rootHttpRouter: coreServices.rootHttpRouter,
    },
    async factory({ rootHttpRouter }) {
      // performs all registration etc inside create
      return DefaultRootHealthService.create({
        rootHttpRouter,
        getLiveness: async () => ...,
      });
    }
  })
)
```

Reply:
of course, this is all moot once we have service modules :)

Reply:
But yeah I see that we did the same type of thing for the auth and lifecycle middlewares in `httpRouterServiceFactory`. So maybe this is where we were going with all this anyway.

Reply:
I think this approach simplifies the customization process for adopters and provides a more opinionated method for presenting the metrics